packages/dashboard/src/components/resourceExplorer/useBuildProvider.ts

```diff
@@ -0,0 +1,43 @@
+import { useState, useEffect } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { ErrorDetails, TreeQuery } from '@iot-app-kit/core';
+import { BranchReference, SiteWiseAssetTreeNode } from '@iot-app-kit/source-iotsitewise';
+import { AssetSummary } from '@aws-sdk/client-iotsitewise';
+import { MaybeSiteWiseAssetTreeSessionInterface } from './types';
+
+export const parseAssetNodes = (assetNodes: SiteWiseAssetTreeNode[]) => {
+  const parsed: AssetSummary[] = [];
+  if (!assetNodes) return parsed;
+  try {
+    for (const node in assetNodes) {
+      parsed.push(assetNodes[node].asset);
+    }
+  } catch (err) {
+    console.log(err);
+  }
+  return parsed;
+};
+
+export const useBuildProvider = (query: TreeQuery<SiteWiseAssetTreeNode[], BranchReference, void>) => {
+  const [provider, setProvider] = useState<MaybeSiteWiseAssetTreeSessionInterface>(undefined);
+  const [errors, setErrors] = useState<ErrorDetails[]>([]);
+  const [rootItems, setRootItems] = useState<AssetSummary[] | undefined>(undefined);
+
+  useEffect(() => {
+    const nextProvider = query.build(uuidv4());
```

Comment:
is there any reason we would need to track the session of this query?

Reply:
Not sure what you mean

Reply:
I remember seeing session id as part of the build interface. Wasn't sure if that was important for any cleanup or reusing the same session if we update the component?

Reply:
I was a little confused on that too. In the docs for the chart components, it says `(Optional) The ID of the widget. A widget is a visualization that you use the [insert name here] chart component to create` -- but the resource explorer isn't a widget, and there's no explanation in the resource explorer docs on why you would want to provide a widgetId, so I left it out for now. We can always add it in later easily if that was an oversight.

Reply:
👍 sounds good!